indico/modules/rb/controllers/backend/bookings.py

```diff
@@ -422,6 +423,31 @@ def _process(self):
         return jsonify(occurrence=reservation_occurrences_schema.dump(self.occurrence, many=False))
 
 
+class RHBookingOccurrenceLinkActions(RHBookingBase):
+    """Links a Reservation occurrence to an event."""
+
+    def _process_args(self):
+        RHBookingBase._process_args(self)
+        date = dateutil.parser.parse(request.view_args['date'], yearfirst=True).date()
+        self.occurrence = self.booking.occurrences.filter(ReservationOccurrence.date == date).one()
+        self.event = Event.query.filter_by(id=request.view_args['event_id'], is_deleted=False).first_or_404()
+
+    def _check_access(self):
+        RHBookingBase._check_access(self)
+        if self.occurrence.link is not None:
+            raise Forbidden('Booking occurrence is already linked')
+        if not self.occurrence.can_link(session.user):
+            raise Forbidden('This user cannot link the event')
+        if not (self.event.start_dt <= server_to_utc(self.occurrence.start_dt) <= self.event.end_dt and
+                self.event.start_dt <= server_to_utc(self.occurrence.end_dt) <= self.event.end_dt):
```

Comment:
Is this an overlap check? If yes better use the `overlaps` util from `indico/util/date_time.py`. Overlap checks are really hard to get your head around, so using the util from which we know it does the right thing makes reviewing (and later reading/understanding) this code easier

Reply:
Yes, it is. It's now changed.